snake/src/main/scala/io/circe/yaml/parser/parse.scala

```diff
@@ -0,0 +1,8 @@
+package io.circe.yaml.parser
+
+import io.circe.yaml.snake
+import io.circe.{Json, ParsingFailure}
+
+object parse extends Parser {
```

Comment:
So it looks nice to put it in the `package.scala` file, but I'm wondering if it makes sense to make it a `val` when the `print` object is both a `Function1` and `Function2` and making that an anonymous class seems a little clunky. It would seem for the sake of consistency to make sense as an object.

The alternative would be to split `print` into 2 functions. Maybe `print` and `printCustom`?